src/components/sub-components/mgt-flyout/mgt-flyout.ts

```diff
@@ -27,33 +28,88 @@ export class MgtFlyout extends LitElement {
     return styles;
   }
 
+  /**
+   * Gets or sets whether the flyout is light dismissable.
+   *
+   * @type {boolean}
+   * @memberof MgtFlyout
+   */
+  @property({
+    attribute: 'light-dismiss',
+    type: Boolean
+  })
+  public isLightDismiss: boolean;
+
   /**
    * Gets or sets whether the flyout is visible
    *
    * @type {string}
-   * @memberof MgtComponent
+   * @memberof MgtFlyout
    */
   @property({
     attribute: 'isOpen',
     type: Boolean
   })
-  public isOpen: boolean;
+  public get isOpen(): boolean {
+    return this._isOpen;
+  }
+  public set isOpen(value: boolean) {
+    if (this._isOpen !== value) {
+      this._isOpen = value;
+
+      if (value) {
+        this.registerLightDismissEvents();
+      } else {
+        this.unregisterLightDismissEvents();
+      }
 
-  private renderedOnce = false;
+      this.requestUpdate('isOpen');
+
+      if (!value) {
+        this.fireCustomEvent('onclose');
+      }
+    }
+  }
+
+  private _isOpen: boolean;
+
+  constructor() {
+    super();
+    this.handleWindowEvent = this.handleWindowEvent.bind(this);
+  }
 
   /**
-   * Invoked when the element is first updated. Implement to perform one time
-   * work on the element after update.
+   * Show the flyout.
+   */
+  public show(): void {
+    this.isOpen = true;
+  }
+
+  /**
+   * Close the flyout.
+   */
+  public close(): void {
+    this.isOpen = false;
+  }
+
+  /**
+   * Invoked each time the custom element is appended into a document-connected element
    *
-   * Setting properties inside this method will trigger the element to update
-   * again after this update cycle completes.
+   * @memberof MgtFlyout
+   */
+  public connectedCallback() {
+    super.connectedCallback();
+    this.registerLightDismissEvents();
```

Comment:
no need to register events unless the flyout is open

Reply:
Oops!